### PR TITLE
added initial apikey endpoint

### PR DESCRIPTION
**Pull request by brngylni**
This PR adds the apikey endpoint #6

### CODE REVIEW

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:11]: Please leave two lines of difference between each segregated block definition.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:15]: Get rid of the added tab please.

```suggestion
    Used for creating a new service by sending a post request to /service/ path.
```

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:21]: Please leave a maximum of one line within the function definition for clarity purposes.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:22]: Add exception handling for `KeyError` (when the `username` key is not found in the JSON body). Abort the request with the `HTTP 429 Unprocessable Entity` error.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:26]: Add exception handling for `KeyError` (when the `valid_till` key is not found in the JSON body). Abort the request with the `HTTP 429 Unprocessable Entity` error.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:27]: Add exception handling for `KeyError` (when the `name` key is not found in the JSON body). Abort the request with the `HTTP 429 Unprocessable Entity` error.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:37]: Use this empty line ;-)

If docstrings are created for a function of a kind, please ensure that they are created for other functions of that kind.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:39]: Add exception handling for `KeyError` (when the `username` key is not found in the JSON body). Abort the request with the `HTTP 429 Unprocessable Entity` error.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:51]: Use this empty line ;-)

If docstrings are created for a function of a kind, please ensure that they are created for other functions of that kind.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:56]: Add exception handling for `KeyError` (when the `apikey_uuid` key is not found in the JSON body). Abort the request with the `HTTP 429 Unprocessable Entity` error.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:65]: More than two lines are left empty before the function definition.

Just two lines should be enough.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:67]: If docstrings are created for a function of a kind, please ensure that they are created for other functions of that kind.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:69]: Add exception handling for `KeyError` (when the `username` key is not found in the JSON body). Abort the request with the `HTTP 429 Unprocessable Entity` error.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:70]: Add exception handling for `KeyError` (when the `apikey_uuid` key is not found in the JSON body). Abort the request with the `HTTP 429 Unprocessable Entity` error.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:86]: Leave one empty line at the end of the file.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/apikey.py:79]: Ref. https://github.com/fedora-infra/webhook-to-fedora-messaging/pull/24#discussion_r1667957399

File: webhook_to_fedora_messaging/endpoints/util.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/util.py:2]: Please leave two lines of difference between each segregated block definition.

File: webhook_to_fedora_messaging/endpoints/util.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/util.py:9]: Please leave two lines of difference between each segregated block definition.

File: webhook_to_fedora_messaging/endpoints/util.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/util.py:12]: Please leave two lines of difference between each segregated block definition.

File: webhook_to_fedora_messaging/endpoints/util.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/util.py:15]: Please leave two lines of difference between each segregated block definition.

File: webhook_to_fedora_messaging/endpoints/util.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/util.py:18]: Please leave two lines of difference between each segregated block definition.

File: webhook_to_fedora_messaging/endpoints/util.py
Comment (gridhead) [inline webhook_to_fedora_messaging/endpoints/util.py:21]: Please leave two lines of difference between each segregated block definition.

File: webhook_to_fedora_messaging/endpoints/apikey.py
Comment (gridhead) [reply webhook_to_fedora_messaging/endpoints/apikey.py:105]: Please do not mark comments as resolved - especially when the suggested change is not addressed. Instead of leaving five lines at the end of the line - please leave just one line.

**Review by gridhead — CHANGES_REQUESTED** [review]
Once you are done with addressing the newly suggested changes, please squash the commits into just one. Every other change can be added after this is merged.

**Comment by brngylni** [comment]
> Once you are done with addressing the newly suggested changes, please squash the commits into just one. Every other change can be added after this is merged.

should I squash all commits so far into one or should I squash them into 1 after every review?

**Comment by gridhead** [comment]
@brngylni Once you get an LGTM from either my end or from @abompard, you can proceed with the squash.

**Review by gridhead — CHANGES_REQUESTED** [review]
Please update the branch to ensure that you keep up with the rebasing merge and once that is done, only include the codebase relevant to this pull request here. Finally, squash the commits and let me know in the channel so I can review it again.

**Comment by brngylni** [comment]
> Please update the branch to ensure that you keep up with the rebasing merge and once that is done, only include the codebase relevant to this pull request here. Finally, squash the commits and let me know in the channel so I can review it again.

How do I fix the interfering commits like in this branch?

**Pull request closed without merging**
